### PR TITLE
Downstream deployedBlockNumber to waku-rln-contract

### PR DESCRIPTION
**Pull request by rymnc**
Downstreams https://github.com/vacp2p/rln-contract/pull/27 and https://github.com/vacp2p/rln-contract/pull/26 to this repo

### CODE REVIEW

**Comment by rymnc** [comment]
Not sure what's going on with CI - 
image
Tests pass - 
```
rymnc@vanguard-work waku-rln-contract % yarn test:foundry                                    
yarn run v1.22.19
$ forge test
[⠆] Compiling...
No files changed, compilation skipped

Running 4 tests for test/WakuRln.t.sol:WakuRlnTest
[PASS] test__Constants() (gas: 7043)
[PASS] test__InvalidFeatures() (gas: 28199)
[PASS] test__InvalidRegistration__Duplicate() (gas: 44397)
[PASS] test__ValidRegistration(uint256[]) (runs: 256, μ: 116690, ~: 73018)
Test result: ok. 4 passed; 0 failed; 0 skipped; finished in 10.18s

Running 11 tests for test/WakuRlnRegistry.t.sol:WakuRlnRegistryTest
[PASS] test__InvalidRegistration_Duplicate(uint256[]) (runs: 256, μ: 2152770, ~: 2061168)
[PASS] test__InvalidRegistration__NoStorageContract() (gas: 9223372036854754743)
[PASS] test__InvalidSingleRegistration__Duplicate(uint256) (runs: 256, μ: 624059, ~: 624059)
[PASS] test__InvalidSingleRegistration__FullTree() (gas: 9355)
[PASS] test__InvalidSingleRegistration__NoStorageContract(uint256) (runs: 256, μ: 593976, ~: 593976)
[PASS] test__NewStorage() (gas: 584289)
[PASS] test__Register(uint256[]) (runs: 256, μ: 716318, ~: 649216)
[PASS] test__RegisterStorage_BadImpl() (gas: 2794447)
[PASS] test__RegisterStorage_BadIndex() (gas: 1145685)
[PASS] test__SingleRegistration(uint256) (runs: 256, μ: 620130, ~: 620130)
[PASS] test__forceProgression() (gas: 608764)
Test result: ok. 11 passed; 0 failed; 0 skipped; finished in 20.22s
Ran 2 test suites: 15 tests passed, 0 failed, 0 skipped (15 total tests)
✨  Done in 23.64s.
```

File: foundry.toml [hunk 1]
```diff
@@ -7,6 +7,4 @@ cache_path  = 'cache_forge'
 
 # 5667: Unused function parameter. Remove or comment out the variable name to silence this warning.
 # 5740: Unreachable code.
-ignored_error_codes = [5667, 5740]
-# setting this to allow expensive larger inserts during tests
-memory_limit = 100_000_000
```
Comment (0x-r4bbit) [inline foundry.toml:12]: Any reason this was removed?

File: test/WakuRlnRegistry.t.sol [hunk 2]
```diff
@@ -56,47 +58,50 @@ contract WakuRlnRegistryTest is Test {
...
     function test__InvalidSingleRegistration__Duplicate(uint256 commitment) public {
+        vm.assume(isValidCommitment(commitment, poseidonHasher.Q()));
         wakuRlnRegistry.newStorage();
         wakuRlnRegistry.register(0, commitment);
         vm.expectRevert(DuplicateIdCommitment.selector);
         wakuRlnRegistry.register(0, commitment);
     }
 
     function test__InvalidSingleRegistration__FullTree() public {
-        vm.pauseGasMetering();
```
Comment (0x-r4bbit) [inline test/WakuRlnRegistry.t.sol:77]: Curious: any reason these gas metering calls have been removed?

**Review by 0x-r4bbit — APPROVED** [review]
Changes LGTM!
Just some small comments I'd like your feedback on :)

File: foundry.toml (hunk 1 above)
Comment (rymnc) [reply foundry.toml:12]: yessir, the tests were grossly inefficient, forcing me to increase the memory more than the default. fixed the issue though, by mocking a revert!

File: test/WakuRlnRegistry.t.sol (hunk 2 above)
Comment (rymnc) [reply test/WakuRlnRegistry.t.sol:77]: same as above, we had to pause metering because of the OutOfGas EvmError

Comment (0x-r4bbit) [reply test/WakuRlnRegistry.t.sol:77]: Ohh, that's interesting!  Thank you!

**Pull request merged**